src/OrbitGl/TimelineUiTest.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <random>
+
+#include "GlCanvas.h"
+#include "MockBatcher.h"
+#include "MockTextRenderer.h"
+#include "MockTimelineInfo.h"
+#include "TimelineUi.h"
+#include "Viewport.h"
+
+namespace orbit_gl {
+
+class TimelineUiTest : public TimelineUi {
+ public:
+  inline static const int kBigScreenPixelsWidth = 2000;
+  inline static const int kSmallScreenMaxPixelsWidth = 700;
+  inline static const int kTinyScreenMaxPixelsWidth = 200;
+
+  explicit TimelineUiTest(int width, MockTimelineInfo* mock_timeline_info, Viewport* viewport,
+                          TimeGraphLayout* layout)
+      : TimelineUi(nullptr /*parent*/, mock_timeline_info, viewport, layout),
+        primitive_assembler_(&mock_batcher_),
+        viewport_(viewport),
+        mock_timeline_info_(mock_timeline_info) {
+    mock_timeline_info_->SetWorldWidth(width);
+    viewport_->SetWorldSize(width, TimelineUi::GetHeight());
```

Comment:
Nit/pers pref: Now that its set up this way, you can probably remove the `width` parameter and the 2 lines
```
 mock_timeline_info_->SetWorldWidth(width);
 viewport_->SetWorldSize(width, TimelineUi::GetHeight());
``` 
and do that initialization before calling the ctor. Aka, do this in `TestUpdatePrimitivesWithSeveralRanges`
```
  MockTimelineInfo mock_timeline_info{world_width};
  Viewport viewport(world_width, TimelineUI::GetHeight());
```

Reply:
TimelineUi::GetHeight() is not an static method. But I can remove the width parameter. What do you think now?